src/components/PostDetails.tsx

```diff
@@ -2,105 +1,122 @@
+import React, { useState, useEffect } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { deleteComment, getPostComments, postComment } from '../api/comments';
+import { Comment, CommentData } from '../types/Comment';
+
+type Props = {
+  post: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [hasError, setHasError] = useState('');
+  const { id, title, body } = post;
+
+  const [showForm, setShowForm] = useState(false);
+
+  useEffect(() => {
+    if (!post.id) {
+      return;
+    }
+
+    setLoading(true);
+    setShowForm(false);
+
+    getPostComments(post.id)
+      .then(setComments)
+      .catch(() => setHasError('Something went wrong'))
+      .finally(() => setLoading(false));
+  }, [post]);
+
+  const addComment = (newComment: CommentData & { postId: number }) => {
+    return postComment(newComment).then(createdComment => {
+      setComments(curr => [...curr, createdComment]);
+    });
+  };
+
+  const removeComment = (commentId: number) => {
+    return deleteComment(commentId).then(() =>
+      setComments(curr => curr.filter(comment => comment.id !== commentId)),
+    );
```

Comment:
Consider adding error handling for the `removeComment` function to manage potential errors from the `deleteComment` API call.